packages/bsky/src/api/app/bsky/subscription/getSubscriptionFeatures.ts

```diff
@@ -0,0 +1,61 @@
+import { Server } from '../../../../lexicon'
+import AppContext from '../../../../context'
+import { Features } from '../../../../lexicon/types/app/bsky/subscription/getSubscriptionFeatures'
+
+export default function (server: Server, ctx: AppContext) {
+  server.app.bsky.subscription.getSubscriptionFeatures({
+    auth: ctx.authVerifier.standard,
+    handler: async ({ auth }) => {
+      const viewer = auth.credentials.iss
+
+      const features = await getFeaturesForViewerSubscription(viewer, ctx)
+
+      return {
+        encoding: 'application/json',
+        body: {
+          features,
+        },
+      }
+    },
+  })
+}
+
+const getFeaturesForViewerSubscription = async (
+  viewerDid: string,
+  ctx: AppContext,
+): Promise<Features> => {
+  const { subscriptionEntitlements } =
+    await ctx.dataplane.getSubscriptionEntitlement({ dids: [viewerDid] })
+
+  const gatedFeatures: Features = {
+    customProfileColor: false,
+  }
+
+  if (subscriptionEntitlements?.length === 0) {
+    return gatedFeatures
+  }
+
+  const { entitlements } = subscriptionEntitlements[0]
+
+  return entitlements.reduce((acc, entitlement) => {
```

Comment:
i totally get what you're doing here, and i'm not really opposed to it, but while we only have one subscription tier, i'm a bit more inclined to make it super straightforward and basically say

```ts
if(entitlements.includes('core') {
  return {
    customProfileColor: false
  }
} else {
  return {
    customProfileColor: true
  }
}
```